File: server_integration_tests/java/java_server_reactive/src/main/java/dev/graphlink/javaserver/impl/AuthorServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.AuthorService;
import dev.graphlink.javaserver.generated.types.Author;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class AuthorServiceImpl implements AuthorService {

    @Override
    public Mono<Author> getAuthor(String id) {
        return Mono.justOrEmpty(Data.authors.stream().filter(a -> a.getId().equals(id)).findFirst());
    }

    @Override
    public Flux<Author> listAuthors() {
        return Flux.fromIterable(Data.authors);
    }
}

File: server_integration_tests/kotlin/kotlin_server_suspend/src/main/kotlin/dev/graphlink/kotlinserversuspend/impl/AuthorServiceImpl.kt
package dev.graphlink.kotlinserversuspend.impl

import dev.graphlink.kotlinserversuspend.generated.services.AuthorService
import dev.graphlink.kotlinserversuspend.generated.types.Author
import org.springframework.stereotype.Service

@Service
class AuthorServiceImpl : AuthorService {
    override suspend fun getAuthor(id: String): Author? = Data.authors.firstOrNull { it.id == id }

    override suspend fun listAuthors(): List<Author> = Data.authors.toList()
}
